components/oneapi-validation/src/main/java/com/wso2telco/dep/oneapivalidation/service/impl/location/ValidateLocation.java

```diff
@@ -31,43 +37,71 @@
     /** The validation rules. */
     private final String[] validationRules = {"queries","location"};
 
+    static Logger logger = Logger.getLogger(ValidateLocation.class);
+
     /* (non-Javadoc)
      * @see com.wso2telco.oneapivalidation.service.IServiceValidate#validate(java.lang.String[])
      */
-    public void validate(String[] params) throws CustomException {
-        //Send parameters within String array according to following order, String[] params = "registrationId", "maxBatchSize";
-        String address = "";
-        Double requestAccuracy = 0.0;
-        try {
-            address = nullOrTrimmed(params[0]);
-            requestAccuracy = Double.parseDouble(params[1]);
-        } catch (ArrayIndexOutOfBoundsException e) {
-            throw new CustomException("SVC0002", null, new String[]{"Missing mandatory parameter address and/or requestedAccuracy"});
-        } catch (NumberFormatException msg){
-            throw new CustomException("SVC0002", null, new String[]{"requestAccuracy"});
-        }
-        
-        ValidationRule[] rules = {
-                new ValidationRule(ValidationRule.VALIDATION_TYPE_MANDATORY_TEL, "address", address),
-                new ValidationRule(ValidationRule.VALIDATION_TYPE_MANDATORY_LOC_ACCURACY, "requestedAccuracy", requestAccuracy)};
-        
-        Validation.checkRequestParams(rules);
-        
-    }
+	public void validate(String[] params) throws CustomException {
+		List<String> addresses = new ArrayList<String>();
+		Double requestAccuracy = 0.0;
```

Comment:
Any reason to keep this as an Object Type. Unless specifically needed, its better to use primitive types.

Reply:
Addressed review comment and fixed